korge/src/commonMain/kotlin/com/soywiz/korge3d/tween/tween.kt

```diff
@@ -0,0 +1,304 @@
+@file:Suppress("NOTHING_TO_INLINE")
+
+package com.soywiz.korge3d.tween
+
+import com.soywiz.kds.iterators.*
+import com.soywiz.klock.*
+import com.soywiz.klock.hr.*
+import com.soywiz.kmem.*
+import com.soywiz.korge3d.component.*
+import com.soywiz.korge3d.View3D
+import com.soywiz.korim.color.*
+import com.soywiz.korio.async.*
+import com.soywiz.korma.geom.*
+import com.soywiz.korma.interpolation.*
+import kotlinx.coroutines.*
+import kotlin.coroutines.*
+import kotlin.jvm.*
+import kotlin.reflect.*
+
+class TweenComponent(
+        override val view: View3D,
+        private val vs: List<V2<*>>,
+        val time: HRTimeSpan = HRTimeSpan.NIL,
+        val easing: Easing = DEFAULT_EASING,
+        val callback: (Double) -> Unit,
+        val c: CancellableContinuation<Unit>
+) : UpdateComponentV2 {
```

Comment:
This whole file is duplicated. I think we should try to figure out a way to deduplicate it. Maybe we can create something like BaseView interface that is implemented by View and View3D? Or extract the functionality in a place that could be called in both cases